File: src/main/java/it/contrader/dto/CategoryDTO.java
package it.contrader.dto;

import it.contrader.model.Tool;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor

public class CategoryDTO {

	private Tool tool;
	
	private Long id;

	private String name;

	private String description;

	
	
	
	
	
}

File: src/dto/categorydto.ts
import { RootContextFlags } from '@angular/core/src/render3/interfaces/view';
import { ToolDTO } from './tooldto';

/**
 * Classe DTO di User. DEVE essere uguale (stesso nome classe, stessi attributi e stessi nomi) a
 * quello nel backend. 
 * 
 * @see Usertype
 * 
 * @author Vittorio Valent
 */
export class CategoryDTO {

    tool: ToolDTO;

    id: number;

    name: string;

    description: string;
}